phoenix-core/src/it/java/org/apache/phoenix/end2end/index/MutableIndexIT.java

```diff
@@ -785,9 +785,51 @@ private void testUpsertingDeletedRowShouldGiveProperDataWithIndexes(boolean mult
           assertEquals(0.5F, rs.getFloat(1), 0.0);
           assertEquals("foo", rs.getString(3));
       }
-  }
+    }
+
+    @Test
+    public void testUpsertingDeletedRowWithNullCoveredColumn() throws Exception {
+        testUpsertingDeletedRowWithNullCoveredColumn(false);
+    }
 
-  /**
+    @Test
+    public void testUpsertingDeletedRowWithNullCoveredColumnMultiCfs() throws Exception {
+        testUpsertingDeletedRowWithNullCoveredColumn(true);
+    }
+
+    public void testUpsertingDeletedRowWithNullCoveredColumn(boolean multiCf) throws Exception {
+        String tableName = "TBL_" + generateUniqueName();
+        String indexName = "IDX_" + generateUniqueName();
+        String columnFamily1 = "cf1";
+        String columnFamily2 = "cf2";
+        String fullTableName = SchemaUtil.getTableName(TestUtil.DEFAULT_SCHEMA_NAME, tableName);
+        String fullIndexName = SchemaUtil.getTableName(TestUtil.DEFAULT_SCHEMA_NAME, indexName);
+        try (Connection conn = getConnection()) {
+            conn.createStatement()
+                    .execute("create table " + fullTableName + " (id integer primary key, "
+                            + (multiCf ? columnFamily1 : "") + "f varchar, "
```

Comment:
Should not this read : 
+ (multiCf ? columnFamily1 + ".": "") + "f varchar, "

if so, the remaining references should change too.

Reply:
Updated the test to fix this.